ci/promote_major_minor_rc.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+set -e
+
+# Script to promote a major/minor RC to stable release
+# Usage: promote_major_minor_rc.sh <rc_tag>
+# Example: promote_major_minor_rc.sh v1.3.0-rc.2
+
+RC_TAG=${1:?"Error: RC tag required (e.g., v1.3.0-rc.2)"}
+TAG_PREFIX=${2:-"v"}
+
+readonly SELF_DIR=$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+
+echo "Promoting RC tag ${RC_TAG} to stable release"
+
+# Parse version from RC tag (v1.3.0-rc.2 -> 1.3.0)
+RC_VERSION=$(echo "${RC_TAG}" | sed "s/^${TAG_PREFIX}//")
+STABLE_VERSION=$(echo "${RC_VERSION}" | sed 's/-rc\.[0-9]*$//')
+
+echo "Stable version will be: ${STABLE_VERSION}"
+
+# Get the original commit that was used to create the RC
+# The RC tag points to a commit with RC version
+# Its parent is the original commit with beta version
+RC_COMMIT=$(git rev-list -n 1 "${RC_TAG}")
+ORIGINAL_COMMIT=$(git rev-list -n 1 "${RC_COMMIT}^")
```

Comment:
Why not just create the release off of the RC commit? That seems like it would be more direct.